common/src/main/java/com/scottlogic/deg/common/profile/constraints/atomic/IsInSetConstraint.java

```diff
@@ -18,45 +18,49 @@
 
 import com.scottlogic.deg.common.profile.Field;
 
-import com.scottlogic.deg.common.profile.RuleInformation;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.DistributedSet;
 
 import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class IsInSetConstraint implements AtomicConstraint {
     public final Field field;
-    public final Set<Object> legalValues;
+    public final DistributedSet<Object> legalValues;
 
-    public IsInSetConstraint(Field field, Set<Object> legalValues) {
+    public IsInSetConstraint(Field field, DistributedSet<Object> legalValues) {
         this.field = field;
         this.legalValues = legalValues;
 
-        if (legalValues.isEmpty()) {
+        if (legalValues.distributedSet().isEmpty()) {
             throw new IllegalArgumentException("Cannot create an IsInSetConstraint for field '" +
                 field.name + "' with an empty set.");
         }
 
-        if (legalValues.contains(null)){
+        if (legalValues.set().stream().anyMatch(Objects::isNull)){
             throw new IllegalArgumentException("Cannot create an IsInSetConstraint for field '" +
                 field.name + "' with a set containing null.");
         }
     }
 
+    public Set<Object> legalValuesWithoutFrequency() {
```

Comment:
this method doesn't need to exist

Reply:
It's used in `FieldSpecFactory` to create a blacklist. This was in an attempt to make the frequency distributions internal to the set, rather than be exposed. This hit a roadblock when looking at merging two distributed sets.

I think this makes the intent clearer - `x.legalValuesWithoutFrequency` vs `x.legalValues.set()` makes it more obvious that we are dropping the weights.